streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -871,7 +871,11 @@ public void process(final ProcessorSupplier<? super K, ? super V> processorSuppl
                                               final Grouped<KR, V> grouped) {
         Objects.requireNonNull(selector, "selector can't be null");
         Objects.requireNonNull(grouped, "grouped can't be null");
-        final GroupedInternal<KR, V> groupedInternal = new GroupedInternal<>(grouped);
+        GroupedInternal<KR, V> groupedInternal = new GroupedInternal<>(grouped);
+        if (groupedInternal.valueSerde() == null) {
```

Comment:
This if for 2): we should pass the serde to Grouped which can then be used if it was not specified, ditto below.